src/common/AdaptiveStream.cpp

```diff
@@ -61,36 +61,48 @@ void AdaptiveStream::ResetSegment()
     absolute_position_ = current_rep_->current_segment_->range_begin_;
 }
 
-bool AdaptiveStream::download_segment()
+unsigned int AdaptiveStream::download_segment()
 {
   if (download_url_.empty())
-    return false;
+    return 400;
 
   return download(download_url_.c_str(), download_headers_);
 }
 
+bool AdaptiveStream::retry(unsigned int returnCode)
+{
+  if (returnCode == 200)
+    return false;
+
+  //Don't retry subtitle segment downloads on HTTP 415 Unsupported Media Type
```

Comment:
My only remark is to better document why we do this, and what we expect as behaviour.

So what is needed here is the following rationale/behaviour:
- A subtitle track is available from the manifest
- The files do not currently exist (HTTP 415) but may exist in the future
- So we do not need to retry the current segment
- But we do want to get the next segment

Reply:
There are more codes that could be an ok response (206 for example)
Maybe just check if the response code starts with 2